pkg/util/admissioncheck/admissioncheck.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissioncheck
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+var (
+	ErrNilParametersRef = errors.New("missing parameters reference")
+	ErrBadParametersRef = errors.New("bad parameters reference")
+)
+
+type objAsPtr[T any] interface {
+	client.Object
+	*T
+}
+
+type ConfigHelper[PtrT objAsPtr[T], T any] struct {
+	gk     schema.GroupKind
+	client client.Client
+}
+
+func NewConfigHelper[PtrT objAsPtr[T], T any](c client.Client) (*ConfigHelper[PtrT, T], error) {
+	helper := ConfigHelper[PtrT, T]{
+		client: c,
+	}
+
+	gvk, err := apiutil.GVKForObject(helper.newConfigPtr(), c.Scheme())
+	if err != nil {
+		return nil, err
+	}
+
+	helper.gk = gvk.GroupKind()
+	return &helper, nil
+}
+
+func (ch *ConfigHelper[PtrT, T]) newConfigPtr() PtrT {
+	return PtrT(new(T))
+}
+
+// IsValidConfigReference - checks if the provided reference is addressing an object specific to this helper.
+func (ch *ConfigHelper[PtrT, T]) IsValidConfigReference(ref *kueue.AdmissionCheckParametersReference) (bool, error) {
+	return refValidForGK(ref, ch.gk)
+}
+
+// ConfigByName - get the config identified by its name
+func (ch *ConfigHelper[PtrT, T]) ConfigByName(ctx context.Context, name string) (PtrT, error) {
+	configPtr := ch.newConfigPtr()
+	if err := ch.client.Get(ctx, types.NamespacedName{Name: name}, configPtr); err != nil {
+		return nil, err
+	}
+	return configPtr, nil
+}
+
+// ConfigFromRef - get the config identified by ref if valid.
+func (ch *ConfigHelper[PtrT, T]) ConfigFromRef(ctx context.Context, ref *kueue.AdmissionCheckParametersReference) (PtrT, error) {
```

Comment:
No need to export this function

Reply:
let's make `IsValidConfigReference` private instead.